native/src/line_token_collection.rs

```diff
@@ -0,0 +1,83 @@
+use crate::comment_block::CommentBlock;
+use crate::line_tokens::LineToken;
+
+#[derive(Debug, Clone)]
+pub struct LineTokenCollection {
+    contents: Vec<LineToken>,
+}
+
+fn insert_at<T>(idx: usize, target: &mut Vec<T>, input: &mut Vec<T>) {
+    let drain = input.drain(..);
+    let mut idx = idx;
+    for item in drain {
+        target.insert(idx, item);
+        idx += 1;
+    }
```

Comment:
Every call to insert has to potentially realloc, and then copy all elements after it. What do you think of something like this:

```suggestion
    let mut target_tail = input.split_off(idx);
    target.append(input);
    target.append(&mut target_tail);
```